src/ucs/sys/topo.h

```diff
@@ -0,0 +1,87 @@
+/**
+* Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_TOPO_H
+#define UCS_TOPO_H
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <ucs/type/status.h>
+#include <limits.h>
+
+BEGIN_C_DECLS
+
+/** @file topo.h */
+
+/**
+ * @ingroup UCS_RESOURCE
+ * System Device abstraction
+ */
+typedef struct ucs_sys_device {
+    unsigned int id;        /**< Index of the unit */
+    unsigned int numa_node; /**< NUMA node assoicated with the device*/
+    const char *bus_id;     /**< bus ID of of the device if applicable*/
+    const char *name;       /**< Name of the device*/
+} ucs_sys_device_t;
+
+
+/*
+ * Capture the estimated latency, bandwidth, and paths to traverse between two
+ * ucs_sys_device_t type structures
+ */
+typedef struct ucs_sys_dev_distance {
+    double latency;          /**< in seconds */
+    double bandwidth;        /**< in bytes/second */
+} ucs_sys_dev_distance_t;
+
+
+/**
+ * Find system device by pci bus id
+ *
+ * @param [in]     bus_id  bus id of the device of interest
+ * @param [in/out] sys_dev pointer to ucs_sys_device_t
+ *                         populated with device associated with the bus_id
+ *
+ * @return UCS_OK or error in case device cannot be found
+ */
+ucs_status_t ucs_topo_find_device_by_pci_id(const char *bus_id, const ucs_sys_device_t *sys_dev);
+
+
+/**
+ * Find system device by device name
+ *
+ * @param [in]     name    char array populated with the name of device of interest
+ * @param [in/out] sys_dev pointer to ucs_sys_device_t
+ *                         populated with device associated with the name array
+ *
+ * @return UCS_OK or error in case device cannot be found
+ */
+ucs_status_t ucs_topo_find_device_by_name(const char *name, const ucs_sys_device_t *sys_dev);
+
+
+/**
+ * Find the distance between two system devices (in terms of latency, bandwidth, hops, etc) 
+ *
+ * @param [in]     device1 ucs_sys_device_t pointing to the first device
+ * @param [in]     device2 ucs_sys_device_t pointing to the second device
+ * @param [in/out] result  pointer to ucs_sys_dev_distance_t
+ *                         populated with distance details between the two
+ *                         devices
+ */
+ucs_status_t ucs_topo_get_distance(const ucs_sys_device_t *device1, const ucs_sys_device_t *device2,
+                                   ucs_sys_dev_distance_t *result);
+
+
+/**
+ * Print a map indicating the topology information between system devices discovered
+ */
+void ucs_topo_print_info(FILE *stream);
```

Comment:
let's stub implementation and a unit test calling ucs_topo_print_info(). just to make sure everything compiles.